sqlglot/parser.py

```diff
@@ -1039,6 +1097,147 @@ def _parse_property_assignment(self, exp_class: t.Type[exp.Expression]) -> exp.E
             this=self._parse_var_or_string() or self._parse_number() or self._parse_id_var(),
         )
 
+    def _parse_properties(self, before=None) -> t.Optional[exp.Expression]:
+        properties = []
+
+        while True:
+            if before:
+                self._match(TokenType.COMMA)
+                identified_property = self._parse_property_before()
+            else:
+                identified_property = self._parse_property()
+
+            if not identified_property:
+                break
+            for p in ensure_collection(identified_property):
+                properties.append(p)
+
+        if properties:
+            return self.expression(exp.Properties, expressions=properties, before=before)
+
+        return None
+
+    def _parse_fallback(self, no=False) -> exp.Expression:
+        self._match_text_seq("FALLBACK")
+        return self.expression(
+            exp.FallbackProperty, no=no, protection=self._match_text_seq("PROTECTION")
+        )
+
+    def _parse_withjournaltable(self) -> exp.Expression:
+        self._match_text_seq("WITH", "JOURNAL", "TABLE")
+        self._match(TokenType.EQ)
+        return self.expression(exp.WithJournalTableProperty, this=self._parse_table_parts())
+
+    def _parse_log(self, no=False) -> exp.Expression:
+        self._match_text_seq("LOG")
+        return self.expression(exp.LogProperty, no=no)
+
+    def _parse_journal(self, no=False, dual=False, before=False) -> exp.Expression:
+        before = self._match_text_seq("BEFORE")
+        self._match_text_seq("JOURNAL")
+        return self.expression(exp.JournalProperty, no=no, dual=dual, before=before)
+
+    def _parse_afterjournal(self, no=False, dual=False, local=None) -> exp.Expression:
+        self._match_text_seq("NOT")
+        self._match_text_seq("LOCAL")
+        self._match_text_seq("AFTER", "JOURNAL")
+        return self.expression(exp.AfterJournalProperty, no=no, dual=dual, local=local)
+
+    def _parse_checksum(self) -> exp.Expression:
+        self._match_text_seq("CHECKSUM")
+        self._match(TokenType.EQ)
+
+        on = None
+        if self._match(TokenType.ON):
+            on = True
+        elif self._match_text_seq("OFF"):
+            on = False
```

Comment:
If we had `match_text_seq` return `None` instead of `False`, this could be more clean.

```suggestion
        on = not self._match_text_seq("OFF") or self._match(TokenType.ON)
```